### DISCUSSION TITLE
Mantine css not work in product version after update nextjs 14.0.1 to 14.3.2

### DISCUSSION THREAD

**Discussion by eurus1** [mantinedev/mantine#6277] [Q&A]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.9.2

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

All

### Describe the bug

i use nextjs 14.1.3 and mantine componnents in my project and project work fine, but after update my nextjs version to 14.2.3 my styles not work in build wersion but when i use the dev mod everyting is ok. can help me to fix this issue?

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rosaldo260** [comment]
Yeah, I see the same issue that mantine styles over-ride my custom styles and it worked before with the same setup.

I've not found a fix for it yet.

**Comment by rtivital** [comment]
What exactly does not work?

If you need to override styles, see https://mantine.dev/styles/mantine-styles/#css-layers

**Reply by eurus1** [reply]
thanks for answer, in this images you can see what happen exactly.
![ok](https://github.com/mantinedev/mantine/assets/42107581/61287c17-8d56-416a-ab32-c4261a8a5409)
![error](https://github.com/mantinedev/mantine/assets/42107581/97aeff2f-ada0-4ce8-aca8-5a4291d71507)

**Comment by rosaldo260** [comment]
I've found the issue - I've imported a client component which uses `Mantine` into the `layout.html` and that causes some components to force mantine styles over the modules. For me it's solved issue, sorry for the not very useful comment.

**Reply by kishoremongar** [reply]
Thank you for sharing your insights.

> It seems like moving the imports to the client component, resolves this conflicts. 
> The imports earlier was called in the server component and was working fine, but has some issues from migrating next.js 14.1 to 14.2

**Comment by PooneJabbari** [comment]
i have the same problem now. ive recently updated nextjs to 15. and mantine to 7.17. everything is okay on development. but some css module styles are not applied in production. is there a fix for it?

**Reply by rtivital** [reply]
See https://help.mantine.dev/q/styles-order
